plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  ActionQueueStatus,
+  ReviewResponseDecision,
+  ReviewStatus,
+} from '../../../src/generated/graphql'
+import { ActionPluginOutput, ActionPluginType } from '../../types'
+import databaseMethods from './databaseMethods'
+import { action as changeStatus } from '../../action_change_status/src'
+
+type TriggeredBy = 'REVIEW' | 'APPLICATION'
+export interface Review {
+  reviewId: number
+  reviewAssignmentId: number
+  applicationId: number
+  reviewerId: number
+  levelNumber: number
+  assignedSections: string[]
+  reviewStatus: ReviewStatus
+}
+interface ChangedApplicationResponse {
+  applicationResponseId: number
+  templateElementId: number
+}
+
+interface ChangedReviewResponse {
+  reviewResponseId: number
+  reviewResponseDecision: ReviewResponseDecision
+  templateElementId: number
+}
+
+type ChangedResponse = ChangedApplicationResponse | ChangedReviewResponse
+
+const updateReviewsStatuses: ActionPluginType = async ({
+  parameters,
+  applicationData,
+  DBConnect,
+}) => {
+  const db = databaseMethods(DBConnect)
+
+  // Get application/reviewId from applicationData if not provided in parameters
+  const applicationId = parameters?.applicationId ?? applicationData?.applicationId
+  const reviewId = parameters?.reviewId ?? applicationData?.reviewData?.reviewId
+  const stageNumber = parameters?.stageNumber || applicationData?.stageNumber
+
+  // Changed responses normally come from the output of the "trimResponses"
+  // action
+  const changedResponses: ChangedResponse[] = parameters.changedResponses || []
+
+  const triggeredBy: TriggeredBy = parameters.triggeredBy || 'APPLICATION'
+
+  console.log(
+    `Updating statuses of reviews associated with ${
+      triggeredBy === 'REVIEW' ? 'review Id: ' + reviewId : 'application Id: ' + applicationId
+    }`
+  )
+
+  try {
+    // Get all reviews for this application/stage. We do additional filtering
+    // (e.g for review level) as required further on.
+    const reviews = await db.getAssociatedReviews(applicationId, stageNumber)
+
+    // APPLICATION SUBMISSIONS:
+    if (triggeredBy === 'APPLICATION') {
+      // Get section codes of the changed responses
+      const changedSections = await db.getChangedSections(
+        changedResponses.map(({ templateElementId }) => templateElementId)
+      )
+
+      // Filter out reviews that are not level 1 and don't intersect with
+      // changed section codes
+      const reviewsToUpdate = reviews.filter(
+        ({ reviewStatus, levelNumber, assignedSections }) =>
+          levelNumber === 1 &&
+          reviewStatus === ReviewStatus.Submitted &&
```

Comment:
Just a though on this specific part - what if we have a Reviewer that is reviewing Section 2 (but still on DRAFT) and then after it was sent to Applicant for RFI there was a new version for one of the responses done by the applicant (even though it wasn't required in the RFI the applicant can change other responses too). Should we also change that Review (which wouldn't be SUBMITTED) to be Pending as well?

Reply:
I think this would already cover that case, as they'd all be included in the application's "changedResponses", right?